### DISCUSSION TITLE
Fluid fields- add ability to set a limit on how many times a field can be used in a given Fluid Field entry.

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#2954] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

The use case where it came up- we have a grid field, we only want one grid with multiple rows, but users keep adding multiple grid fields instead.

You can specify limits in Grid & Relationships as a way to confine the quantity of rows/relationships and this would be great to have on the Fluid Field also.

**Comment by intoeetive** [comment]
I think we need to start with cnaging the UI of how the fields are assigned into Fluid.

That way, we can allow setting the fluid "blocks" as:
- required
- limited to single or multiple usage

We might use layout similar to https://github.com/ExpressionEngine/ExpressionEngine/pull/3555

@TomJaeger

**Reply by intoeetive** [reply]
Also https://github.com/ExpressionEngine/ExpressionEngine/discussions/2941 probably falls into same scope

**Comment by intoeetive** [comment]
The UI might look similar to this: 
![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/752126/576cfb50-7252-4dfa-98bd-e0e646f016d1)

We'd need to make sure the list becomes searchable (and preferably scrollable) when there are many fields. Clicking pencil allows editing the name for the field as it would appear in the Fluid
